server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -108,6 +110,8 @@
   // null and nonnull multiple times as stop() and start() are called.
   @Nullable
   private volatile ConcurrentHashMap<String, DruidDataSource> dataSources = null;
```

Comment:
This field is now unused?

Reply:
it was being used, but not anymore with updates via the snapshot.